benchmark.cc

```diff
@@ -0,0 +1,187 @@
+#include "Product.h"
+#include "Chained.h"
+#include "Open.h"
+#include "StdMap.h"
+#include "StdUnorderedMap.h"
+
+#include <iostream>
+#include <vector>
+#include <chrono>
+#include <unistd.h>
+
+enum MAP_TYPE {STDMAP = 0, CHAINEDMAP = 1, OPENMAP = 2, STDUNORDEREDMAP = 3};
+
+void read_args(int, char**);
+
+//command line options
+MAP_TYPE mapType = STDMAP;
+uint64_t verbose = 0;
+uint64_t mapSize = 100000;
+double load = 0.7;
+
+struct productsArr{
+    std::string* products;
+    uint64_t size;
+};
+
+struct graphStat{
+    graphStat(MAP_TYPE t, double l, uint64_t n) {
+        mapType = t;
+        loadFactor = l;
+        numberOfItems = n;
+        msToFill = -1;
+        msToRead = -1;
+    }
+    MAP_TYPE mapType;
+    double loadFactor;
+    uint64_t numberOfItems;
+    uint64_t msToFill;
+    uint64_t msToRead;
+
+    std::string getType(){
+        switch (mapType) {
+            case STDMAP:
+                return "STD Map";
+            case CHAINEDMAP:
+                return "Chained Map";
+            case OPENMAP:
+                return "Open Map";
+            case STDUNORDEREDMAP:
+                return "STD Unordered Map";
+        }
+        return "";
+    }
+};
+
+bool compareMsFill(graphStat a, graphStat b) {
+    return a.msToFill < b.msToFill;
+}
+
+bool compareMsRead(graphStat a, graphStat b) {
+    return a.msToRead < b.msToRead;
+}
+
+graphStat testBuildTime(const uint64_t mapSize, MAP_TYPE mapType, double loadFactor){
+    auto time1 = std::chrono::high_resolution_clock::now();
+    auto time2 = std::chrono::high_resolution_clock::now();
+    HashTable<uint64_t, uint64_t>* map;
+    switch(mapType){
+        case CHAINEDMAP:
+            map = new ChainedMap<uint64_t, uint64_t>(loadFactor);
```

Comment:
Don't benchmark memory allocation if you can help it.  Allocate everything first, then start the timer.